diff.go

```diff
@@ -259,95 +285,158 @@ func (a *App) diffGroups(
 		}
 	}
 
-	return groupDiff{
+	return &groupDiff{
 		groupsToCreate:  groupsToCreate,
 		groupsToUpdate:  groupsToUpdate,
 		groupsToRemove:  groupsToRemove,
 		membersToAdd:    membersToAdd,
 		membersToRemove: membersToRemove,
-	}
+	}, nil
 }
 
 type usersDiff struct {
 	create []YtsaurusUser
 	update []UpdatedYtsaurusUser
 	remove []YtsaurusUser
+	result map[ObjectID]YtsaurusUser
 }
 
 func (a *App) diffUsers(
-	azureUsersMap map[AzureID]AzureUser,
-	ytUsersMap map[AzureID]YtsaurusUser,
-) usersDiff {
+	sourceUsers []SourceUser,
+	ytUsers []YtsaurusUser,
+) (*usersDiff, error) {
+	sourceUsersMap := make(map[ObjectID]SourceUser)
+	for _, user := range sourceUsers {
+		sourceUsersMap[user.GetID()] = user
+	}
+
+	ytUsersMap := make(map[ObjectID]YtsaurusUser)
+	resultUsersMap := make(map[ObjectID]YtsaurusUser)
+	for _, user := range ytUsers {
+		sourceUser, err := a.buildSourceUser(&user)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to build source user")
+		}
+		ytUsersMap[sourceUser.GetID()] = user
+		resultUsersMap[sourceUser.GetID()] = user
+	}
+
+	ytUsersFromSourceMap := make(map[ObjectID]YtsaurusUser)
+	for _, user := range sourceUsers {
+		ytUser, err := a.buildYtsaurusUser(user)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create Ytsaurus user from source user")
+		}
+		ytUsersFromSourceMap[user.GetID()] = ytUser
+	}
```

Comment:
I think ytUsersFromSourceMap can be safely removed now, I don't see usages anymore